web_timeline/static/src/js/web_timeline.js

```diff
@@ -0,0 +1,514 @@
+/*---------------------------------------------------------
+ * Odoo web_timeline
+ * Copyright 2015 ACSONE SA/NV
+ * Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+ *---------------------------------------------------------*/
+
+_.str.toBoolElse = function (str, elseValues, trueValues, falseValues) {
+    var ret = _.str.toBool(str, trueValues, falseValues);
+    if (_.isUndefined(ret)) {
+        return elseValues;
+    }
+    return ret;
+};
+
+odoo.define('web_timeline.TimelineView', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var form_common = require('web.form_common');
+    var Model = require('web.DataModel');
+    var time = require('web.time');
+    var View = require('web.View');
+    var widgets = require('web_calendar.widgets');
+
+    var _t = core._t;
+    var _lt = core._lt;
+    var QWeb = core.qweb;
+
+    function isNullOrUndef(value) {
+        return _.isUndefined(value) || _.isNull(value);
+    }
+
+    var TimelineView = View.extend({
+        template: "TimelineView",
+        display_name: _lt('Timeline'),
+        icon: 'fa-clock-o',
+        quick_create_instance: widgets.QuickCreate,
+
+        init: function (parent, dataset, view_id, options) {
+            this._super(parent);
+            this.ready = $.Deferred();
+            this.permissions = {};
+            this.set_default_options(options);
+            this.dataset = dataset;
+            this.model = dataset.model;
+            this.fields_view = {};
+            this.view_id = view_id;
+            this.view_type = 'timeline';
+            this.color_map = {};
+            this.range_start = null;
+            this.range_stop = null;
+            this.selected_filters = [];
+            this.current_window = null;
+        },
+
+        get_perm: function(name){
+            var self = this;
+            var promise = self.permissions[name];
+            if(!promise) {
+                var defer = $.Deferred();
+                new Model(this.dataset.model)
+                .call("check_access_rights", [name, false])
```

Comment:
Isn't this redundant? You can assume any call without permissions will fail by itself, not?